From dc06656820b1f3b188bf3115a1098280ecc293b8 Mon Sep 17 00:00:00 2001
From: Tomasz Wasilczyk <twasilczyk@google.com>
Date: Tue, 26 Jan 2021 08:46:31 -0800
Subject: [PATCH] Don't crash if default supervision profile owner is not set

Bug: 175430552
Bug: 197399948
Test: build, flash, watch logcat

Change-Id: If441946fa278c04ae88122f6243f5a7dedd96ebc
Merged-In: If441946fa278c04ae88122f6243f5a7dedd96ebc
(cherry picked from commit bd27c365915f824824d865668c63a2119c4700fb)
(cherry picked from commit bcb5ee6f43ea24fb74649f164ca3c3032a7be3b5)
(cherry picked from commit b5fa0a6c5e96c420c1f6d808be603c4579f9a1ba)
Merged-In:If441946fa278c04ae88122f6243f5a7dedd96ebc
---
 .../server/devicepolicy/DevicePolicyManagerService.java        | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java b/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
index 13ea6baf5a6..a7452e88d9b 100644
--- a/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
+++ b/services/devicepolicy/java/com/android/server/devicepolicy/DevicePolicyManagerService.java
@@ -9381,6 +9381,9 @@ public class DevicePolicyManagerService extends BaseIDevicePolicyManager {
             final ComponentName doComponent = mOwners.getDeviceOwnerComponent();
             final ComponentName poComponent =
                     mOwners.getProfileOwnerComponent(userHandle.getIdentifier());
+            if (supervisorComponent == null) {
+                return null;
+            }
             if (supervisorComponent.equals(doComponent) || supervisorComponent.equals(
                     poComponent)) {
                 return supervisorComponent;
-- 
2.35.0.263.gb82422642f-goog

